lib/project_types/script/commands/enable.rb

```diff
@@ -41,6 +45,28 @@ def self.help
       def self.extended_help
         ShopifyCli::Context.message('script.enable.extended_help', ShopifyCli::TOOL_NAME)
       end
+
+      private
+
+      def acquire_configuration(config_file: nil, config_props: nil)
+        properties = {}
+        properties = YAML.load(File.read(config_file)) unless config_file.nil?
```

Comment:
Just curious what happens if file doesn't exist or if it exists but has invalid format?  What errors are displayed?

Reply:
We get a syntax error:
```
Traceback (most recent call last):
	27: from /Users/spencerhaan/src/github.com/Shopify/shopify-app-cli/bin/shopify:30:in `<main>'
	26: from /Users/spencerhaan/src/github.com/Shopify/shopify-app-cli/vendor/deps/cli-kit/lib/cli/kit/error_handler.rb:21:in `call'
	25: from /Users/spencerhaan/src/github.com/Shopify/shopify-app-cli/vendor/deps/cli-kit/lib/cli/kit/error_handler.rb:75:in `handle_abort'
	24: from /Users/spencerhaan/src/github.com/Shopify/shopify-app-cli/bin/shopify:31:in `block in <main>'
	23: from /Users/spencerhaan/src/github.com/Shopify/shopify-app-cli/lib/shopify-cli/core/entry_point.rb:33:in `call'
	22: from /Users/spencerhaan/src/github.com/Shopify/shopify-app-cli/lib/shopify-cli/core/monorail.rb:20:in `log'
	21: from /Users/spencerhaan/src/github.com/Shopify/shopify-app-cli/lib/shopify-cli/core/entry_point.rb:34:in `block in call'
	20: from /Users/spencerhaan/src/github.com/Shopify/shopify-app-cli/lib/shopify-cli/core/executor.rb:17:in `call'
	19: from /Users/spencerhaan/src/github.com/Shopify/shopify-app-cli/vendor/deps/cli-kit/lib/cli/kit/executor.rb:14:in `call'
	18: from /Users/spencerhaan/src/github.com/Shopify/shopify-app-cli/vendor/deps/cli-kit/lib/cli/kit/executor.rb:45:in `with_traps'
	17: from /Users/spencerhaan/src/github.com/Shopify/shopify-app-cli/vendor/deps/cli-kit/lib/cli/kit/executor.rb:57:in `twrap'
	16: from /Users/spencerhaan/src/github.com/Shopify/shopify-app-cli/vendor/deps/cli-kit/lib/cli/kit/executor.rb:46:in `block in with_traps'
	15: from /Users/spencerhaan/src/github.com/Shopify/shopify-app-cli/vendor/deps/cli-kit/lib/cli/kit/executor.rb:57:in `twrap'
	14: from /Users/spencerhaan/src/github.com/Shopify/shopify-app-cli/vendor/deps/cli-kit/lib/cli/kit/executor.rb:47:in `block (2 levels) in with_traps'
	13: from /Users/spencerhaan/src/github.com/Shopify/shopify-app-cli/vendor/deps/cli-kit/lib/cli/kit/executor.rb:15:in `block in call'
	12: from /Users/spencerhaan/src/github.com/Shopify/shopify-app-cli/vendor/deps/cli-kit/lib/cli/kit/executor.rb:37:in `with_logging'
	11: from /Users/spencerhaan/src/github.com/Shopify/shopify-app-cli/vendor/deps/cli-ui/lib/cli/ui.rb:171:in `log_output_to'
	10: from /Users/spencerhaan/src/github.com/Shopify/shopify-app-cli/vendor/deps/cli-kit/lib/cli/kit/executor.rb:38:in `block in with_logging'
	 9: from /Users/spencerhaan/src/github.com/Shopify/shopify-app-cli/vendor/deps/cli-ui/lib/cli/ui/stdout_router.rb:169:in `with_id'
	 8: from /Users/spencerhaan/src/github.com/Shopify/shopify-app-cli/vendor/deps/cli-kit/lib/cli/kit/executor.rb:39:in `block (2 levels) in with_logging'
	 7: from /Users/spencerhaan/src/github.com/Shopify/shopify-app-cli/vendor/deps/cli-kit/lib/cli/kit/executor.rb:17:in `block (2 levels) in call'
	 6: from /Users/spencerhaan/src/github.com/Shopify/shopify-app-cli/lib/shopify-cli/command.rb:22:in `call'
	 5: from /Users/spencerhaan/src/github.com/Shopify/shopify-app-cli/lib/project_types/script/commands/enable.rb:25:in `call'
	 4: from /Users/spencerhaan/src/github.com/Shopify/shopify-app-cli/lib/project_types/script/commands/enable.rb:53:in `acquire_configuration'
	 3: from /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/psych.rb:277:in `load'
	 2: from /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/psych.rb:390:in `parse'
	 1: from /System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/psych.rb:456:in `parse_stream'
/System/Library/Frameworks/Ruby.framework/Versions/2.6/usr/lib/ruby/2.6.0/psych.rb:456:in `parse': (<unknown>): could not find expected ':' while scanning a simple key at line 2 column 1 (Psych::SyntaxError)
```

We could catch any errors during the load and re-throw it without the stack? I wasn't sure what the best way to handle this would be or if I should just let it bubble up.

I also tried messing with other things like non-string values, empty or misformatted properties for `--config_props`, etc, and they all return a error saying the input was invalid:
```
Can't enable script. An error was returned: Variable configuration of type ConfigurationInput was provided invalid value for entries.0.value (Expected value to not be null), entries.1.value (Could not coerce value 12 to String).
```

Additionally, if the file doesn't exist we get another stacktrace and an error about the file not existing:
```
Traceback (most recent call last):
	24: from /Users/spencerhaan/src/github.com/Shopify/shopify-app-cli/bin/shopify:30:in `<main>'
	23: from /Users/spencerhaan/src/github.com/Shopify/shopify-app-cli/vendor/deps/cli-kit/lib/cli/kit/error_handler.rb:21:in `call'
	22: from /Users/spencerhaan/src/github.com/Shopify/shopify-app-cli/vendor/deps/cli-kit/lib/cli/kit/error_handler.rb:75:in `handle_abort'
	21: from /Users/spencerhaan/src/github.com/Shopify/shopify-app-cli/bin/shopify:31:in `block in <main>'
	20: from /Users/spencerhaan/src/github.com/Shopify/shopify-app-cli/lib/shopify-cli/core/entry_point.rb:33:in `call'
	19: from /Users/spencerhaan/src/github.com/Shopify/shopify-app-cli/lib/shopify-cli/core/monorail.rb:20:in `log'
	18: from /Users/spencerhaan/src/github.com/Shopify/shopify-app-cli/lib/shopify-cli/core/entry_point.rb:34:in `block in call'
	17: from /Users/spencerhaan/src/github.com/Shopify/shopify-app-cli/lib/shopify-cli/core/executor.rb:17:in `call'
	16: from /Users/spencerhaan/src/github.com/Shopify/shopify-app-cli/vendor/deps/cli-kit/lib/cli/kit/executor.rb:14:in `call'
	15: from /Users/spencerhaan/src/github.com/Shopify/shopify-app-cli/vendor/deps/cli-kit/lib/cli/kit/executor.rb:45:in `with_traps'
	14: from /Users/spencerhaan/src/github.com/Shopify/shopify-app-cli/vendor/deps/cli-kit/lib/cli/kit/executor.rb:57:in `twrap'
	13: from /Users/spencerhaan/src/github.com/Shopify/shopify-app-cli/vendor/deps/cli-kit/lib/cli/kit/executor.rb:46:in `block in with_traps'
	12: from /Users/spencerhaan/src/github.com/Shopify/shopify-app-cli/vendor/deps/cli-kit/lib/cli/kit/executor.rb:57:in `twrap'
	11: from /Users/spencerhaan/src/github.com/Shopify/shopify-app-cli/vendor/deps/cli-kit/lib/cli/kit/executor.rb:47:in `block (2 levels) in with_traps'
	10: from /Users/spencerhaan/src/github.com/Shopify/shopify-app-cli/vendor/deps/cli-kit/lib/cli/kit/executor.rb:15:in `block in call'
	 9: from /Users/spencerhaan/src/github.com/Shopify/shopify-app-cli/vendor/deps/cli-kit/lib/cli/kit/executor.rb:37:in `with_logging'
	 8: from /Users/spencerhaan/src/github.com/Shopify/shopify-app-cli/vendor/deps/cli-ui/lib/cli/ui.rb:171:in `log_output_to'
	 7: from /Users/spencerhaan/src/github.com/Shopify/shopify-app-cli/vendor/deps/cli-kit/lib/cli/kit/executor.rb:38:in `block in with_logging'
	 6: from /Users/spencerhaan/src/github.com/Shopify/shopify-app-cli/vendor/deps/cli-ui/lib/cli/ui/stdout_router.rb:169:in `with_id'
	 5: from /Users/spencerhaan/src/github.com/Shopify/shopify-app-cli/vendor/deps/cli-kit/lib/cli/kit/executor.rb:39:in `block (2 levels) in with_logging'
	 4: from /Users/spencerhaan/src/github.com/Shopify/shopify-app-cli/vendor/deps/cli-kit/lib/cli/kit/executor.rb:17:in `block (2 levels) in call'
	 3: from /Users/spencerhaan/src/github.com/Shopify/shopify-app-cli/lib/shopify-cli/command.rb:22:in `call'
	 2: from /Users/spencerhaan/src/github.com/Shopify/shopify-app-cli/lib/project_types/script/commands/enable.rb:25:in `call'
	 1: from /Users/spencerhaan/src/github.com/Shopify/shopify-app-cli/lib/project_types/script/commands/enable.rb:53:in `acquire_configuration'
/Users/spencerhaan/src/github.com/Shopify/shopify-app-cli/lib/project_types/script/commands/enable.rb:53:in `read': No such file or directory @ rb_sysopen - config2.yml (Errno::ENOENT)
```

Reply:
Would it be worth spinning the error handling fixes into a subsequent ticket just so this isn't blocked for the preview?

Reply:
OK, let's create another ticket for now.  And document it as a known issue.

Reply:
https://github.com/Shopify/script-service/issues/1485